_data-prepper/managing-data-prepper/source-coordination.md

```diff
@@ -0,0 +1,148 @@
+| `sourcePartitionKey` | String  | The identifier for the partition of work associated with this item. For example, for an `s3` source with scan capabilities, this identifier is the S3 bucket's `objectKey` combination.
+| `partitionOwner` | String   | An identifier for the node that actively owns and is working on this partition. This ID contains the hostname of the node, but is `null` when this partition is not owned. |
+| `partitionProgressState` | String  | A JSON string object representing the progress made on a partition of work, or any additional metadata that may be needed by the source in the case of another node picking up where the last node left off during a crash.  |
+| `partitionOwnershipTimeout` | Timestamp  | Whenever a Data Prepper node acquires a partition, a 10 minute timeout is given to the owner of the partition to handle the case of a node crashing. The ownership is renewed with another 10 minutes when the owner saves state of the partition.  |
+| `sourcePartitionStatus` | Enum | Represents the current state of the partition: `ASSIGNED` means the partition is currently being processed. `UNASSIGNED` means the partition is waiting to be processed. `CLOSED` means the partition is waiting to be processed at a later date;. Lastly, `COMPLETED` means the partition is already processed. |           
+| `reOpenAt` | Timestamp  | Represents the time that CLOSED partitions re-open and are considered available for processing. Only applies to CLOSED partitions. |
+| `closedCount` | Long | Tracks how many times the partition has been marked as `CLOSED`.|
+
+
+## Acquiring partitions
+
+Partitions are acquired in the order that they are returned in the `List<PartitionIdentifer>` provided by the source. When a node attempts to acquire a partition, Data Prepper performs the following steps.
+
+1.  Query the `ASSIGNED` partitions to check if any of these `ASSIGNED` partitions have expired partition owners. This is intended to give priority to partitions that had nodes crash in the middle of processing, which can allow for using a partition state that may be time sensitive. 
+2. After querying `ASSIGNED` partitions, Data Prepper queries the `CLOSED` partitions to see if any of the partition's `reOpenAt` timestamps have been reached. 
+3. If there are no `ASSIGNED` or `CLOSED` partitions available, then Data Prepper queries the `UNASSIGNED` partitions one of these is partitions is `ASSIGNED`.
+
+If this flow occurs, and no partition is acquired by the node, then the partition supplier function provided in the `getNextPartition` method of `SourceCoordinator` will create new partitions. After the supplier function completes, Data Prepper again queries the partitions for `ASSIGNED`, `CLOSED`, and `UNASSIGNED`.
```

Comment:
Should "the" precede `SourceCoordinator`?

Reply:
I don't think so, since we're referring to the setting and not the concept.